src/mapping/Mapping.cpp

```diff
@@ -9,16 +9,19 @@ namespace precice {
 namespace mapping {
 
 Mapping::Mapping(
-    Constraint constraint,
-    int        dimensions,
-    bool       requireGradient)
```

Comment:
The initialize order still seems messy, or is this a false alert from the CI? The argument order  of the constructor needs to match the order of the class members, correct?

Reply:
I kept the order of the arguments instead of the declaration order, then moved dimensions earlier, I'll clean this up